capstone-backend/src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -100,23 +141,45 @@ public void ensureCustomNumberOfBooksInResponse() {
 
   @Test
   public void checkConvertResponseToVolumeDataGeneratesCorrectQuantityOfObjects() {
-    Collection<VolumeData> books = SearchServlet.convertResponseToVolumeData(artificialJsonResponse);
+    Collection<VolumeData> books = SearchServlet.convertResponseToVolumeData(ARTIFICIAL_JSON_RESPONSE);
     int expectedBookQuantity = 3;
 
     Assert.assertEquals(expectedBookQuantity, books.size());
   }
 
   @Test
   public void checkConvertResponseToVolumeDataPopulatesObjects() {
-    Collection<VolumeData> books = SearchServlet.convertResponseToVolumeData(artificialJsonResponse);
-    
-    boolean anyNotPopulated = books.stream()
-      .anyMatch( book -> book.getID() != null ||
-      book.getTitle() != null || 
-      book.getAuthors() != null ||
-      book.getDescription() != null ||
-      book.getThumbnailLink() != null);
+    Collection<VolumeData> books = SearchServlet.convertResponseToVolumeData(ARTIFICIAL_JSON_RESPONSE);
+
+    boolean anyNotPopulated = books.stream().anyMatch(book -> checkPopulatedVolumeDataObject(book));
```

Comment:
Should this be renamed to `anyPopulated`? Since you're using `anyMatch` it looks like this expression will be true as long as at least one book contains a non-null field. Alternatively, do you want this test to check that **at least one** book is populated, or that **all** books are populated?

Reply:
I changed this to `allPopulated` using `allMatch` instead of `anyMatch` I also refactored `checkPopulatedVolumeDataObject` to return false if there are any default values!

It was intended to check that **all** books are populated with nondefault values.